main.tf

```
@@ -430,6 +430,67 @@ resource "aws_security_group_rule" "computed_ingress_with_self" {
     var.rules[lookup(var.computed_ingress_with_self[count.index], "rule", "_")][2],
   )
 }
+# Security group rules with "prefix_list_ids", but without "cidr_blocks", "self" or "source_security_group_id"
+resource "aws_security_group_rule" "ingress_with_prefix_list_ids" {
+  count = var.create ? length(var.ingress_with_prefix_list_ids) : 0
+
+  security_group_id = local.this_sg_id
+  type              = "ingress"
+
+  prefix_list_ids = var.ingress_prefix_list_ids
```

```suggestion
  prefix_list_ids = var.ingress_with_prefix_list_ids
```
Or something along those lines, right?

Please verify in other resources, too.

---

I'm a little confused here. The `ingress_with_prefix_list_ids` is the list of custom rules to apply, not the list of prefix IDs. I was just following the template of the other ones which create a resource per rule (`ingress_with_prefix_list_ids` and then set the argument to the resource ID to apply to (`ingress_prefix_list_ids`). Let me know if I have confused something and if you could clarify.

---

You're right. I misunderstood the code.